[Cocoa] Static images should show the frame whose index is primary image index

shallawa opened the pull request:
#### d8bc465282794ce0568c9b4c6a932171e3a1bb1f
<pre>
[Cocoa] Static images should show the frame whose index is primary image index
<a href="https://bugs.webkit.org/show_bug.cgi?id=245099">https://bugs.webkit.org/show_bug.cgi?id=245099</a>
rdar://99479666

Reviewed by Darin Adler.

For the static image, the BitmapImage::m_currentFrame should be initialized with
CGImageSourceGetPrimaryImageIndex() after all the encoded data is received.

* LayoutTests/TestExpectations:
* LayoutTests/fast/images/image-overlay-mutiple-frames-expected.html: Added.
* LayoutTests/fast/images/image-overlay-mutiple-frames.html: Added.
* LayoutTests/fast/images/resources/avif_overlay-exact-bounds-last-frame.avif: Added.
* LayoutTests/fast/images/resources/avif_overlay-exact-bounds.avif: Added.
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/platform/graphics/BitmapImage.cpp:
(WebCore::BitmapImage::dataChanged):
(WebCore::BitmapImage::resetAnimation):
* Source/WebCore/platform/graphics/BitmapImage.h:
* Source/WebCore/platform/graphics/ImageDecoder.h:
(WebCore::ImageDecoder::primaryFrameIndex const):
* Source/WebCore/platform/graphics/ImageSource.cpp:
(WebCore::ImageSource::clearMetadata):
(WebCore::ImageSource::primaryFrameIndex):
(WebCore::ImageSource::dump):
* Source/WebCore/platform/graphics/ImageSource.h:
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp:
(WebCore::ImageDecoderCG::primaryFrameIndex const):
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.h:

Canonical link: <a href="https://commits.webkit.org/261009@main">https://commits.webkit.org/261009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/507b5a9364292c018add48fd6a2ed26ee9b56398

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110157 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19255 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42819 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1581 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119157 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114108 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20797 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10444 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102411 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115903 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98630 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43644 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30291 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85483 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11948 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31628 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12561 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8594 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17926 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51244 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7626 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14367 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (changes requested):
Please include a regression test.

shallawa committed:
[Cocoa] Static images should show the frame whose index is primary image index
https://bugs.webkit.org/show_bug.cgi?id=245099
rdar://99479666

Reviewed by Darin Adler.

For the static image, the BitmapImage::m_currentFrame should be initialized with
CGImageSourceGetPrimaryImageIndex() after all the encoded data is received.

* LayoutTests/TestExpectations:
* LayoutTests/fast/images/image-overlay-mutiple-frames-expected.html: Added.
* LayoutTests/fast/images/image-overlay-mutiple-frames.html: Added.
* LayoutTests/fast/images/resources/avif_overlay-exact-bounds-last-frame.avif: Added.
* LayoutTests/fast/images/resources/avif_overlay-exact-bounds.avif: Added.
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/platform/graphics/BitmapImage.cpp:
(WebCore::BitmapImage::dataChanged):
(WebCore::BitmapImage::resetAnimation):
* Source/WebCore/platform/graphics/BitmapImage.h:
* Source/WebCore/platform/graphics/ImageDecoder.h:
(WebCore::ImageDecoder::primaryFrameIndex const):
* Source/WebCore/platform/graphics/ImageSource.cpp:
(WebCore::ImageSource::clearMetadata):
(WebCore::ImageSource::primaryFrameIndex):
(WebCore::ImageSource::dump):
* Source/WebCore/platform/graphics/ImageSource.h:
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp:
(WebCore::ImageDecoderCG::primaryFrameIndex const):
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.h:

Canonical link: https://commits.webkit.org/261009@main
